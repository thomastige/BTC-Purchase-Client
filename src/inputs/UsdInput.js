import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import * as actions from '../actions';

/*
USD Input component. Contains the label and the input form for the USD amount to be converted to BTC.
*/
class UsdInput extends React.Component {

	constructor(props){
		super(props);
		this.props.dispatch(actions.setUsdInput(''));
		this.handleChange = this.handleChange.bind(this);
	}

	handleChange(e){
		if (this.props.btcFxRateReducer != undefined && this.props.btcFxRateReducer != 0) {
			var newValue = e.target.value
			this.props.dispatch(actions.setUsdInput(newValue));
		}
	}
	
	render() {
		return (
			<div>
				<input value="USD" className="inputLabel" id="UsdInputLabel" readOnly/>
				<br/>
				<br/>
				<input value={this.props.usdInput} id="usdInput" onChange={this.handleChange} placeholder={this.props.placeholderText}/>
			</div>
		);
	}
}
function mapDispatchToProps(dispatch) {
	var placeholderText = "Enter your amount";
	if (dispatch.btcFxRateReducer == undefined || dispatch.btcFxRateReducer == 0) {
		placeholderText = "Fetching Exchange Rates...";
	}
	return { 
		placeholderText: placeholderText,
		btcFxRateReducer: dispatch.btcFxRateReducer,
		usdInput: dispatch.usdInputReducer,
		actions: bindActionCreators(actions.setUsdInput, dispatch)
	}
}
export default connect(mapDispatchToProps)(UsdInput);