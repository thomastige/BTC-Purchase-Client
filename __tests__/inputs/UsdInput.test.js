import React from 'react';
import UsdInput from 'src/inputs/UsdInput';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store'


Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureStore();
let store;

test('Testing UsdInput snapshot', () => {
	store = mockStore({});
	const wrapper = shallow(<UsdInput store={store}/>);
    expect(wrapper).toMatchSnapshot();
});