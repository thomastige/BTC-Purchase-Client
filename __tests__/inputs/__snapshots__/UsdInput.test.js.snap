// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UsdInput snapshot 1`] = `
<UsdInput
  actions={[Function]}
  dispatch={[Function]}
  placeholderText="Fetching Exchange Rates..."
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
